tests/integration/states/test_pkg.py

```diff
@@ -29,73 +31,50 @@
 from salt.ext import six
 from salt.ext.six.moves import range  # pylint: disable=import-error,redefined-builtin
 
-try:
-    from distro import LinuxDistribution
-    pre_grains = LinuxDistribution()
-except ImportError:
-    pre_grains = None
-
 log = logging.getLogger(__name__)
 
-_PKG_EPOCH_TARGETS = []
-_PKG_TARGETS = ['figlet', 'sl']
-_PKG_32_TARGETS = []
-_PKG_CAP_TARGETS = []
-_PKG_DOT_TARGETS = []
-_WILDCARDS_SUPPORTED = False
-_VERSION_SPEC_SUPPORTED = True
-
-if salt.utils.platform.is_windows():
-    _PKG_TARGETS = ['7zip', 'putty']
-elif salt.utils.platform.is_freebsd:
-    _VERSION_SPEC_SUPPORTED = False
-elif pre_grains:
-    if any(arch in pre_grains.like() for arch in ('arch', 'archlinux')):
-        _WILDCARDS_SUPPORTED = True
-    elif 'debian' in pre_grains.like():
-        _WILDCARDS_SUPPORTED = True
-    elif 'rhel' in pre_grains.like():
-        _PKG_TARGETS = ['units', 'zsh-html']
-        _WILDCARDS_SUPPORTED = True
-        if pre_grains.id() == 'centos':
-            if pre_grains.major_version() == 5:
-                _PKG_32_TARGETS = ['xz-devel.i386']
-            else:
-                _PKG_32_TARGETS.append('xz-devel.i686')
-        if pre_grains.major_version() == 5:
-            _PKG_DOT_TARGETS = ['python-migrate0.5']
-        elif pre_grains.major_version() == 6:
-            _PKG_DOT_TARGETS = ['tomcat6-el-2.1-api']
-        elif pre_grains.major_version() == 7:
-            _PKG_DOT_TARGETS = ['tomcat-el-2.2-api']
-            _PKG_EPOCH_TARGETS = ['comps-extras']
-    elif pre_grains.id() in ('sles', 'opensuse'):
-        _PKG_TARGETS = ['figlet', 'htop']
-        _PKG_CAP_TARGETS = [('perl(ZNC)', 'znc-perl')]
-
-
-def runs_on(platforms=None, os_like=None, reason=''):
-    def decorator(caller):
-        @functools.wraps(caller)
-        def wrapper(cls):
-            if platforms:
-                if not any(getattr(salt.utils.platform, 'is_' + platform)() for platform in platforms):
-                    cls.skipTest(reason if reason else 'OS not in [{}]'.format(', '.join(platforms)))
-            if pre_grains and os_like:
-                if not any(x in pre_grains.like() for x in os_like):
-                    cls.skipTest(reason if reason else 'OS not similar to [{}]'.format(', '.join(os_like)))
-            return caller(cls)
-
-        return wrapper
-
-    return decorator
-
 
 @destructiveTest
 class PkgTest(ModuleCase, SaltReturnAssertsMixin):
+    _PKG_EPOCH_TARGETS = []
```

Comment:
There's really no reason why these class attribute aren't also set in `setUpClass` right?

Reply:
The `skipIf` decorator doesn't have access to `self`.  These need to be declared outside of `setUpClass` so that certain tests can be skipped if they are empty.

Reply:
Ok